### PR TITLE
Recognize inherited DynamoDBAttributes on properties

### PR DESCRIPTION
**Pull request by pjh0463**
Fix attribute inheritance issue with DynamoDB in object persistence model

## Description
This pull request addresses the issue where `DynamoDBAttribute`s on properties were not being inherited in the object persistence model. The fix involves updating the `GetAttributeObjects` method to use `CustomAttributeExtensions.GetCustomAttributes` (`Attribute.GetCustomAttributes` for .Net Framework) instead of `MemberInfo.GetCustomAttributes`.

## Motivation and Context
This change is required to ensure correct serialization and deserialization of objects when using inheritance in the AWS SDK .NET with DynamoDB. This PR fixes issue #2590.

## Testing
The changes were tested by creating a parent class with a property decorated with a DynamoDBProperty attribute and a child class that inherits from the parent class and overrides the property. The SDK now recognizes the attribute in the property in the inherited class as expected.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

## License
- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/aws/aws-sdk-net/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

**Comment by ashishdhingra** [comment]
As reported by user, per [MemberInfo.GetCustomAttributes Method](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.memberinfo.getcustomattributes?view=net-7.0), `inherit` parameter is ignored for properties and events.

**Review by ashishdhingra — APPROVED** [review]
As reported by user, per [MemberInfo.GetCustomAttributes Method](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.memberinfo.getcustomattributes?view=net-7.0), `inherit` parameter is ignored for properties and events.

Tested locally. LGTM. 🚀

File: sdk/test/Services/DynamoDBv2/UnitTests/Custom/DynamoDBTests.cs [hunk 1]
```diff
@@ -17,6 +17,8 @@
 
 namespace AWSSDK_DotNet35.UnitTests
 {
+    using Amazon.DynamoDBv2;
```
Comment (peterrsongg) [inline sdk/test/Services/DynamoDBv2/UnitTests/Custom/DynamoDBTests.cs:20]: nit: can you put this above `namespace` for consistency?

Comment (pjh0463) [reply sdk/test/Services/DynamoDBv2/UnitTests/Custom/DynamoDBTests.cs]: changed it. thank you for the comment.

**Review by peterrsongg — APPROVED** [review]

**Comment by ashovlin** [comment]
This was manually merged and released today as [version 3.7.103 of AWSSDK.DynamoDBv2](https://www.nuget.org/packages/AWSSDK.DynamoDBv2/3.7.103).

Thanks again for your contribution!

**Pull request closed without merging**